### PR TITLE
Blockwise: stringify constant key inputs

### PR DESCRIPTION
**Pull request by madsbk**
Fixing a bug in https://github.com/dask/dask/pull/7734, which didn't stringify the constant key inputs.

- [ ] Closes https://github.com/dask/distributed/issues/4980
- [ ] Tests added / passed
- [ ] Passes `black dask` / `flake8 dask` / `isort dask`

### CODE REVIEW

**Comment by jsignell** [comment]
I think failing tests are just https://github.com/dask/dask/issues/7840

**Comment by madsbk** [comment]
> I think failing tests are just #7840

Thanks @jsignell, I was also surprised that this PR could affect those tests :)

**Comment by mrocklin** [comment]
Rerunning CI

File: dask/tests/test_distributed.py [hunk 1]
```diff
@@ -546,6 +546,39 @@ def f(d, a):
...
 
 
+def test_map_partitions_df_input():
+    """
+    Check that map_partitions can handle a delayed
+    partition of a dataframe input
+    """
+    pd = pytest.importorskip("pandas")
+    dd = pytest.importorskip("dask.dataframe")
+    dist = pytest.importorskip("distributed")
```
Comment (jrbourbeau) [inline dask/tests/test_distributed.py:556]: We can drop this line as we skip the entire module if `distributed` isn't installed

https://github.com/dask/dask/blob/d2ce17c73918dff45981eb8d9c64f7bbe48fc595/dask/tests/test_distributed.py#L3

File: dask/tests/test_distributed.py [hunk 2]
```diff
@@ -546,6 +546,39 @@ def f(d, a):
...
+
+        df = dd.from_pandas(pd.DataFrame({"a": range(5)}), npartitions=2)
+        df = df.map_partitions(f, dl, meta=df._meta)
+        df = df.persist(optimize_graph=False)
+        df.compute()
+
+    with dist.LocalCluster(
+        scheduler_port=0, asynchronous=False, n_workers=1, nthreads=1, processes=False
+    ) as cluster:
+        with dist.Client(cluster, asynchronous=False):
```
Comment (jrbourbeau) [inline dask/tests/test_distributed.py:578]: Can we drop this section in favor of using `@gen_cluster` like most of the other tests in this module? `gen_cluster` tends to be both lighter on CI and makes for easier debugging

**Review by jrbourbeau — COMMENTED** [review]
Thanks for the fix @madsbk!

File: dask/tests/test_distributed.py (hunk 1 above)
Comment (madsbk) [reply dask/tests/test_distributed.py]: Fixed

File: dask/tests/test_distributed.py (hunk 2 above)
Comment (madsbk) [reply dask/tests/test_distributed.py]: For some reason using `@gen_cluster` doesn't trigger the bug in https://github.com/dask/distributed/issues/4980.

Comment (madsbk) [reply dask/tests/test_distributed.py]: I suspect that something more is wrong regarding Blockwise and delayed objects. This PR is the a first step but I will  investigate.

**Comment by madsbk** [comment]
@jrbourbeau I think we should merge this PR _as is_. I am working on fixing additional issues with Blockwise and remote keys: https://github.com/dask/dask/pull/7849

**Review by jrbourbeau — APPROVED** [review]
Thanks @madsbk!

**Pull request merged**